### PR TITLE
libtraceevent: 1.8.0 -> 1.8.1

### PR DESCRIPTION
**Pull request by trofi**
While at it added a trivial updater.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/linux/libtraceevent/default.nix [hunk 1]
```diff
@@ -23,6 +23,12 @@ stdenv.mkDerivation rec {
   doCheck = true;
   checkInputs = [ cunit ];
 
+  passthru.updateScript = gitUpdater {
+    # No nicer place to find latest release.
+    url = "https://git.kernel.org/pub/scm/libs/libtrace/libtraceevent.git";
+    rev-prefix = "libtraceevent-";
+  };
```
Comment (wentasah) [inline pkgs/os-specific/linux/libtraceevent/default.nix:30]: What's the purpose of the `updateScript` here? AFAICS both r-ryantm and `nix-update` can update the package even without the script.

**Review by wentasah — COMMENTED** [review]
Otherwise LGTM.

Changelog is [here](https://git.kernel.org/pub/scm/libs/libtrace/libtraceevent.git/tag/?h=libtraceevent-1.8.1) and looks harmless. `nixpkgs-review` also didn't found any build problem so far (still waiting for 2 libre kernel builds, but don't expect problems there). `trace-cmd` works.

File: pkgs/os-specific/linux/libtraceevent/default.nix (hunk 1 above)
Comment (trofi) [reply pkgs/os-specific/linux/libtraceevent/default.nix:30]: > `nix-update` can update the package even without the script.

Not for me. I usually try `nix-update P` and `nix-shell maintainers/scripts/update.nix  --argstr package P`.

Both fail for `libtraceevent`:

```
$ nix-update libtraceevent
$ nix eval --json --impure --expr
let

  pkgs = import "/home/slyfox/dev/git/nixpkgs-staging-next";
  args =  builtins.functionArgs pkgs;
  inputs = (if args ? system then { system = builtins.currentSystem; } else {}) //
           (if args ? overlays then { overlays = [ ]; } else {});
  pkg = (pkgs inputs)."libtraceevent";
  sanitizePosition = x: x;

  raw_version_position = sanitizePosition (builtins.unsafeGetAttrPos "version" pkg);

  position = if pkg ? isRubyGem then
    raw_version_position
  else if pkg ? isPhpExtension then
    raw_version_position
   else
    sanitizePosition (builtins.unsafeGetAttrPos "src" pkg);
in {
  name = pkg.name;
  old_version = pkg.version or (builtins.parseDrvName pkg.name).version;
  inherit raw_version_position;
  filename = position.file;
  line = position.line;
  urls = pkg.src.urls or null;
  url = pkg.src.url or null;
  rev = pkg.src.rev or null;
  hash = pkg.src.outputHash or null;
  go_modules = pkg.goModules.outputHash or null;
  go_modules_old = pkg.go-modules.outputHash or null;
  cargo_deps = pkg.cargoDeps.outputHash or null;
  raw_cargo_lock =
    if pkg ? cargoDeps.lockFile then
      let
        inherit (pkg.cargoDeps) lockFile;
        res = builtins.tryEval (sanitizePosition {
          file = toString lockFile;
        });
      in
      if res.success then res.value.file else false
    else
      null;
  npm_deps = pkg.npmDeps.outputHash or null;
  yarn_deps = pkg.offlineCache.outputHash or null;
  tests = builtins.attrNames (pkg.passthru.tests or {});
  has_update_script = pkg.passthru.updateScript or null != null;
  src_homepage = pkg.src.meta.homepage or null;
  changelog = pkg.meta.changelog or null;
} --extra-experimental-features flakes nix-command
warning: unknown experimental feature 'configurable-impure-env'
Traceback (most recent call last):
  File "/nix/store/r15pz72vbp19f8nd8wk18la3ws29fnxl-nix-update-1.0.0/bin/.nix-update-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/r15pz72vbp19f8nd8wk18la3ws29fnxl-nix-update-1.0.0/lib/python3.11/site-packages/nix_update/__init__.py", line 287, in main
    package = update(options)
              ^^^^^^^^^^^^^^^
  File "/nix/store/r15pz72vbp19f8nd8wk18la3ws29fnxl-nix-update-1.0.0/lib/python3.11/site-packages/nix_update/update.py", line 344, in update
    update_hash = update_version(
                  ^^^^^^^^^^^^^^^
  File "/nix/store/r15pz72vbp19f8nd8wk18la3ws29fnxl-nix-update-1.0.0/lib/python3.11/site-packages/nix_update/update.py", line 285, in update_version
    new_version = fetch_latest_version(
                  ^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/r15pz72vbp19f8nd8wk18la3ws29fnxl-nix-update-1.0.0/lib/python3.11/site-packages/nix_update/version/__init__.py", line 132, in fetch_latest_version
    raise VersionError(
nix_update.errors.VersionError: Please specify the version. We can only get the latest version from codeberg/crates.io/gitea/github/gitlab/notabug/pypi/savannah/sourcehut/rubygems projects right now
```

```
$ nix-shell maintainers/scripts/update.nix --argstr package libtraceevent
error:
       … while calling the 'derivationStrict' builtin

         at /derivation-internal.nix:9:12:

            8|
            9|   strict = derivationStrict drvAttrs;
             |            ^
           10|

       … while evaluating derivation 'nixpkgs-update-script'
         whose name attribute is located at /home/slyfox/dev/git/nixpkgs-staging-next/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'shellHook' of derivation 'nixpkgs-update-script'

         at /home/slyfox/dev/git/nixpkgs-staging-next/maintainers/scripts/update.nix:223:3:

          222|   '';
          223|   shellHook = ''
             |   ^
          224|     unset shellHook # do not contaminate nested shells

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Package with an attribute name `libtraceevent` does not have a `passthru.updateScript` attribute defined.
```

After the change `nix-shell maintainers/scripts/update.nix --argstr package libtraceevent` works fine for me. Does `nix-update` work for you as is?

Comment (wentasah) [reply pkgs/os-specific/linux/libtraceevent/default.nix:30]: I see. `nix-update` works without updateScript if you manually specify the version, such as `nix-update libtraceevent --version 1.8.1`. With the updateScript, `nix-update libtraceevent -u` is sufficient.

**Review by wentasah — APPROVED** [review]
In the mean time r-ryantm generated an update in #277950, which made me notice that you have a typo in the commit title :-) If you want to fix it, do it, but I think we can merge it even as it's now.

**Comment by trofi** [comment]
> In the mean time r-ryantm generated an update in #277950, which made me notice that you have a typo in the commit title :-) If you want to fix it, do it, but I think we can merge it even as it's now.

Gah, good catch! Fixed package name typo in PR and commit descriptions.

**Review by wentasah — APPROVED** [review]
LGTM

**Pull request merged**